src/components/organisms/DataTable/DataTable.tsx

```diff
@@ -55,33 +63,49 @@ const DataTable = <TData extends object>(
     tableSize = TableSizeTypes.M,
     title,
     onSortingChange,
-    onColumnFiltersChange,
-    pagination,
-    setPagination,
+    onFiltersChange,
+    paginationData,
+    onPaginationChange,
     meta,
     getSubRows,
     pageSizeOptions,
+    hidePagination = false,
   }: DataTableProps<TData>,
   ref: Ref<HTMLDivElement>
 ) => {
+  const { per_page, total } = paginationData || {}
   const [expanded, setExpanded] = useState<ExpandedState>({})
+  const [pagination, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: per_page || 10,
+  })
+
+  useEffect(() => {
+    if (onPaginationChange) {
+      onPaginationChange({
+        per_page: pagination.pageSize,
+        page: pagination.pageIndex + 1,
+      })
+    }
+  }, [pagination])
 
   const table = useReactTable<TData>({
     data,
     columns,
+    manualPagination: !!paginationData, // Tell react-table that you will handle the pagination manually
+    pageCount: total, // Provide the total number of pages
```

Comment:
**[blocker]**

"total" from BE tells you the total number of items that exist in this list, not the total number of pages. Currently AddedUsersTable thinks we have 2 pages of information and navigating to the second page means that the table disappears.

**Extra**: Would also be good to not hide the entire table, when we have no results, but we know that there should be more than 1 page of information. Will help us avoid some edge cases in the future. (Cases where we have 2 pages of information when rendering, some information is deleted from Database and user will end up with no table, when they try to go to the next page)